reccmp/isledecomp/compare/db.py

```diff
@@ -470,16 +496,16 @@ def match_vtable(
         for_vftable = f"{name}::`vftable'{{for `{for_name}'}}"
 
         # Try to match on the "vftable for X first"
-        recomp_addr = self._find_potential_match(for_vftable, SymbolType.VTABLE)
-        if recomp_addr is not None:
-            return self.set_pair(addr, recomp_addr, SymbolType.VTABLE)
+        for obj in self.search_name(for_vftable, SymbolType.VTABLE):
+            if obj.orig_addr is None and obj.recomp_addr is not None:
+                return self.set_pair(addr, obj.recomp_addr, SymbolType.VTABLE)
```

Comment:
Maybe add a comment here that we intentionally return the first match and ignore others. Same below

Reply:
I added a docstring and swapped the search order around. I always have to remind myself how this function works so hopefully it's way more clear now. Let me know what you think!